Haddocks for cardano chain cannot be produced

We should remove the haddock errors (that might have carried over). Otherwise we cannot produce haddock documentation, and therefore we cannot use `hoogle`.

```
    src/Cardano/Chain/Update/Vote.hs:307:7: error:
        parse error on input ‘-- ^ This byte is part of the signed payload, but is not part of the transmitted payload.
          -- This is an implementation artifact of the previous’
        |
    307 |       -- ^ This byte is part of the signed payload, but is not part of the transmitted payload.
        |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

--- a/scripts/buildkite/rebuild.hs
+++ b/scripts/buildkite/rebuild.hs
@@ -58,7 +58,13 @@ buildStep testArgs = do
  where
   cfg = ["--dump-logs", "--color", "always"]
   stackBuild args = run "stack" $ cfg ++ ["build", "--fast"] ++ args
-  buildArgs    = ["--bench", "--no-run-benchmarks", "--no-haddock-deps"]
+  buildArgs =
+    [ "--bench"
+    , "--no-run-benchmarks"
+    , "--haddock"
+    , "--haddock-internal"
+    , "--no-haddock-deps"
+    ]
   buildAndTest = stackBuild $ ["--tests"] ++ buildArgs
   build        = stackBuild $ ["--no-run-tests"] ++ buildArgs
   test =

